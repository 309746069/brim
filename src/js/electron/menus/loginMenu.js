/* @flow */

import {MenuItem} from "electron"

import {createSearchMenuTemplate} from "./searchMenu"
import {isArray} from "../../lib/is"

export function createLoginMenuTemplate(send: Function) {
  let menu = createSearchMenuTemplate(send)

  disableChildren(menu, "Query")
  disable(menu, "Toggle Search History")
  disable(menu, "Toggle Log Details")

  if (process.platform === "darwin") {
    disable(menu, "Preferences...")
  }

  return menu
}

function disable(menu, label) {
  let item = findByLabel(menu, label)
  item.enabled = false
}

function disableChildren(menu, label) {
  let item = findByLabel(menu, label)
  item.submenu.forEach((i) => (i.enabled = false))
}

function findByLabel(menu, label): MenuItem {
  let item = find(menu, label)
  if (!item) throw `Unknown menu item "${label}"`
  return item
}

function find(item: MenuItem, label): MenuItem {
  if (item.label === label) return item
  if (item.submenu) return find(item.submenu, label)
  if (isArray(item)) {
    for (let i of item) {
      let found = find(i, label)
      if (found) return found
    }
  }
}
