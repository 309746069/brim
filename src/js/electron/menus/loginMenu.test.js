/* @flow */
import {createLoginMenuTemplate} from "./loginMenu"

let send

beforeEach(() => {
  send = jest.fn()
})

test("login menu", () => {
  let template = createLoginMenuTemplate(send)

  expect(template).toMatchSnapshot()
})
