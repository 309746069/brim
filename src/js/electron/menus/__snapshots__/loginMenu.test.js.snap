// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login menu 1`] = `
Array [
  Object {
    "label": "TestApp",
    "submenu": Array [
      Object {
        "role": "about",
      },
      Object {
        "type": "separator",
      },
      Object {
        "click": [Function],
        "enabled": false,
        "label": "Preferences...",
      },
      Object {
        "role": "services",
        "submenu": Array [],
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "hide",
      },
      Object {
        "role": "hideothers",
      },
      Object {
        "role": "unhide",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "quit",
      },
    ],
  },
  Object {
    "label": "Edit",
    "submenu": Array [
      Object {
        "role": "undo",
      },
      Object {
        "role": "redo",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "cut",
      },
      Object {
        "role": "copy",
      },
      Object {
        "role": "paste",
      },
      Object {
        "role": "pasteandmatchstyle",
      },
      Object {
        "role": "delete",
      },
      Object {
        "role": "selectall",
      },
      Object {
        "type": "separator",
      },
      Object {
        "label": "Speech",
        "submenu": Array [
          Object {
            "role": "startspeaking",
          },
          Object {
            "role": "stopspeaking",
          },
        ],
      },
    ],
  },
  Object {
    "label": "Query",
    "submenu": Array [
      Object {
        "accelerator": "CmdOrCtrl+K",
        "click": [Function],
        "enabled": false,
        "label": "Pin Search",
      },
      Object {
        "accelerator": "CmdOrCtrl+Shift+K",
        "click": [Function],
        "enabled": false,
        "label": "Clear Pins",
      },
      Object {
        "accelerator": "CmdOrCtrl+L",
        "click": [Function],
        "enabled": false,
        "label": "Focus Search Bar",
      },
      Object {
        "enabled": false,
        "type": "separator",
      },
      Object {
        "accelerator": "CmdOrCtrl+Left",
        "click": [Function],
        "enabled": false,
        "label": "Back",
      },
      Object {
        "accelerator": "CmdOrCtrl+Right",
        "click": [Function],
        "enabled": false,
        "label": "Forward",
      },
    ],
  },
  Object {
    "label": "View",
    "submenu": Array [
      Object {
        "role": "reload",
      },
      Object {
        "role": "forcereload",
      },
      Object {
        "role": "toggledevtools",
      },
      Object {
        "accelerator": "CmdOrCtrl+i",
        "click": [Function],
        "label": "Toggle Search Inspector",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "resetzoom",
      },
      Object {
        "role": "zoomin",
      },
      Object {
        "role": "zoomout",
      },
      Object {
        "type": "separator",
      },
      Object {
        "accelerator": "CmdOrCtrl+[",
        "click": [Function],
        "enabled": false,
        "label": "Toggle Search History",
      },
      Object {
        "accelerator": "CmdOrCtrl+]",
        "click": [Function],
        "enabled": false,
        "label": "Toggle Log Details",
      },
      Object {
        "type": "separator",
      },
      Object {
        "role": "togglefullscreen",
      },
    ],
  },
  Object {
    "role": "window",
    "submenu": Array [
      Object {
        "role": "minimize",
      },
      Object {
        "role": "close",
      },
      Object {
        "click": [Function],
        "label": "Reset State",
      },
    ],
  },
  Object {
    "role": "help",
    "submenu": Array [
      Object {
        "click": [Function],
        "label": "Learn More",
      },
    ],
  },
]
`;
